src/charm.py

```diff
@@ -530,6 +536,11 @@ def _create_cluster(self) -> None:
         self.app_peer_data["units-added-to-cluster"] = "1"
         self.unit_peer_data["unit-initialized"] = "True"
         self.unit_peer_data["member-role"] = "primary"
+        try:
+            subprocess.check_call(["open-port", "3306/tcp"])
+            subprocess.check_call(["open-port", "33060/tcp"])
+        except subprocess.CalledProcessError:
+            logger.exception("failed to open port")
```

Comment:
should we be going into BlockedStatus if there is an issue opening ports?

Reply:
I have checked several charms, they behave differently:
 * [charm-lxd](https://github.com/canonical/charm-lxd/commit/cc4fdafe0fca2c1b9bcfd32efc03916fdddacaf8#diff-b9ed39bbc9c0387bd3e07da31d13373745534a1cd723d3e292c73496b12e307cR1697) rise logger.error
 * [charm-microk8s](https://github.com/canonical/charm-microk8s/commit/8472b0f862bb98a5f185523b092336e7a2d759e0#diff-b9ed39bbc9c0387bd3e07da31d13373745534a1cd723d3e292c73496b12e307cR234-R239) set maintenance->active
None above are based on operators framework.

@jnsgruk  [zinc-k8s-operator](https://github.com/jnsgruk/zinc-k8s-operator/commit/2076f30311ce915aae7b982de33b03e0b4bb15ce#diff-b9ed39bbc9c0387bd3e07da31d13373745534a1cd723d3e292c73496b12e307cR78-R82) just report error via logger.error

Since it is not a primary charm functionality, I see no reason to block/error the charm.
The error message about failing to expose port is good enough => use `logger.error` like [here](https://github.com/jnsgruk/zinc-k8s-operator/commit/2076f30311ce915aae7b982de33b03e0b4bb15ce#diff-b9ed39bbc9c0387bd3e07da31d13373745534a1cd723d3e292c73496b12e307cR78-R82)